### ISSUE TITLE
`generate_small_primes` doesn't test division by 3

### ISSUE THREAD

**Issue by meithecatte** [prdx23/1024-bit-primes#3]
Running the code shows that some non-primes, such as 9 or 15, [get included in the output](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=67f8e5459316628e52fd7b37401db767).

**Comment by igorcrevar** [comment]
I think correct implementation for `trial_division` method is code bellow (check out `https://en.wikipedia.org/wiki/Primality_test#Simple_methods`)

```
fn trial_division(n: u64, start: u64) -> PrimeResult {
    if n == 2 || n == 3 {
       return PrimeResult::Prime;
    }

    if n <= 1 || n % 2 == 0 || n % 3 == 0 {
       return PrimeResult::Composite;
    }
    // assumption: n > 3 and start > 3
    let root_n = (n as f64).sqrt() as u64;
    for x in (start..(root_n + 1)).step_by(6) {
        if n % x == 0 || n % (x + 2) == 0 {
            return PrimeResult::Composite;
        }
    }
    PrimeResult::Prime
}
```

**Comment by prdx23** [comment]
Hey @meithecatte and @igorcrevar, Thanks for noticing this!

An [earlier version](https://github.com/prdx23/1024-bit-primes/blob/420980735c2f43bb9ee0164e27bf7ae3c1e64ed6/src/algos.rs#L29) of this test used to have a check for `n % 3 == 0` which somehow got removed in later cleanup. `generate_small_primes()` iterates with +=2, and I set the last bit of all random numbers to 1 so this test will not receive even numbers, but I should probably mention that in the assumptions. 

Fortunately this does not affect the primality testing in the final version of the code as it's only used for an early exit, and fixing this would make it run very slightly faster as the few false positives that were reaching miller-rabin would now get removed in the early exit. 

I will update the article and code with this fix.
